scripts/populate_tox/populate_tox.py

```diff
@@ -0,0 +1,502 @@
+
+    # Determine Python support
+    expected_python_versions = TEST_SUITE_CONFIG[integration].get("python")
+    if expected_python_versions:
+        expected_python_versions = SpecifierSet(expected_python_versions)
+    else:
+        expected_python_versions = SpecifierSet(f">={MIN_PYTHON_VERSION}")
+
+    def _supports_lowest(release: Version) -> bool:
+        time.sleep(0.1)  # don't DoS PYPI
+        py_versions = determine_python_versions(fetch_release(package, release))
+        target_python_versions = TEST_SUITE_CONFIG[integration].get("python")
+        if target_python_versions:
+            target_python_versions = SpecifierSet(target_python_versions)
+        return bool(supported_python_versions(py_versions, target_python_versions))
+
+    if not _supports_lowest(releases[0]):
+        i = bisect_left(releases, True, key=_supports_lowest)
+        if i != len(releases) and _supports_lowest(releases[i]):
+            # we found the lowest version that supports at least some Python
+            # version(s) that we do, cut off the rest
+            releases = releases[i:]
+
+    return releases
+
+
+def pick_releases_to_test(releases: list[Version]) -> list[Version]:
+    """Pick a handful of releases from a list of supported releases."""
+    # If the package has majors (or major-like releases, even if they don't do
+    # semver), we want to make sure we're testing them all. If not, we just pick
+    # the oldest, the newest, and a couple in between.
+    has_majors = len(set([v.major for v in releases])) > 1
+    filtered_releases = set()
+
+    if has_majors:
+        # Always check the very first supported release
+        filtered_releases.add(releases[0])
+
+        # Find out the min and max release by each major
+        releases_by_major = {}
+        for release in releases:
+            if release.major not in releases_by_major:
+                releases_by_major[release.major] = [release, release]
+            if release < releases_by_major[release.major][0]:
+                releases_by_major[release.major][0] = release
+            if release > releases_by_major[release.major][1]:
+                releases_by_major[release.major][1] = release
+
+        for i, (min_version, max_version) in enumerate(releases_by_major.values()):
+            filtered_releases.add(max_version)
+            if i == len(releases_by_major) - 1:
+                # If this is the latest major release, also check the lowest
+                # version of this version
+                filtered_releases.add(min_version)
+
+    else:
+        indexes = [
+            0,  # oldest version supported
+            len(releases) // 3,
+            len(releases) // 3 * 2,  # two releases in between, roughly evenly spaced
+            -1,  # latest
+        ]
+
+        for i in indexes:
+            try:
+                filtered_releases.add(releases[i])
+            except IndexError:
```

Comment:
I think we can omit this `try/except`. Unless `releases == []`, I don't think it is possible to have an index error, since `len(releases) // 3` (and also `len(releases)//3 * 2`) should always be between `0` and `len(releases) - 1`

Reply:
If we get rid of the `try/except` I think it might also be more readable to manually add each release, rather than do it in a for loop